### PR TITLE
Refactor and update the logic that deals with the snap-push store endpoint

### PR DESCRIPTION
**Pull request by sergiusens**

### CODE REVIEW

File: snapcraft/storeapi/__init__.py [hunk 1]
```diff
@@ -328,11 +316,90 @@ def register(self, snap_name, series):
...
+
+    def __init__(self, status_details_url):
+        self.__status_details_url = status_details_url
+
+    def track(self):
+        queue = Queue()
+        thread = Thread(target=self._update_status, args=(queue,))
+        thread.start()
+
+        widgets = ['Processing...', AnimatedMarker()]
```
Comment (kyrofa) [inline snapcraft/storeapi/__init__.py]: Would it look better if there was a space between these two?

Comment (sergiusens) [inline snapcraft/storeapi/__init__.py]: I don't know, is this a question or a statement? :-)

Comment (kyrofa) [inline snapcraft/storeapi/__init__.py]: Haha, a question. But you can ignore it, this looks good to me :+1: .

**Comment by sergiusens** [comment]
LP: #1572399 and LP: #1599953

**Pull request merged**